source/hp/refinement.cc

```diff
@@ -828,7 +828,7 @@ namespace hp
       // possible
       using level_type =
         typename dealii::DoFHandler<dim, spacedim>::active_fe_index_type;
-      static const level_type invalid_level =
+      constexpr level_type invalid_level =
```

Comment:
```suggestion
      const level_type invalid_level =
```
I don't see a good reason why this should be `constexpr` or `static` anyway.

Reply:
True. Just make it a regular `const` variable.

But there is a semantic issue here. Why would we assign a variable with name `invalid_active_fe_index` to a variable called `invalid_level` and type `level_type`? These two things are completely different. I *think* what is called for is something like `static_cast<level_type>(-1)` instead.

Reply:
Oh, I'm responsible for this. `static_cast<level_type>(-1)` would indeed be a better choice here.

Since `level_type` is essentially the same as `active_fe_index_type`, the result will be the same. It is more clear with the `static_cast` though.

Reply:
Fixed!